lib/app/services/database/conversation_db_service.c.dart

```diff
@@ -0,0 +1,345 @@
+import 'package:ion/app/features/chat/model/entities/private_direct_message_data.c.dart';
+import 'package:ion/app/features/chat/model/entities/private_message_reaction_data.c.dart';
+import 'package:ion/app/services/database/ion_database.c.dart';
+import 'package:nostr_dart/nostr_dart.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:uuid/uuid.dart';
+
+part 'conversation_db_service.c.g.dart';
+
+@Riverpod(keepAlive: true)
+ConversationsDBService conversationsDBService(Ref ref) =>
+    ConversationsDBService(ref.watch(ionDatabaseProvider));
+
+class ConversationsDBService {
+  ConversationsDBService(this._db);
+
+  final IONDatabase _db;
+
+  Future<int> _insertConversationData({
+    required String conversationId,
+    required EventMessage eventMessage,
+    bool isDeleted = false,
+  }) {
+    final conversationMessage = PrivateDirectMessageEntity.fromEventMessage(eventMessage);
+    return _db.into(_db.conversationMessagesTable).insert(
+          ConversationMessagesTableData(
+            isDeleted: isDeleted,
+            conversationId: conversationId,
+            eventMessageId: conversationMessage.id,
+            createdAt: conversationMessage.createdAt,
+            pubKeys: conversationMessage.allPubkeysMask,
+            subject: conversationMessage.data.relatedSubject?.value,
+            status: DeliveryStatus.none,
+          ),
+          mode: InsertMode.insertOrReplace,
+        );
+  }
+
+  Future<int> _insertConversationReactionsTableData(EventMessage eventMessage) {
+    final reactionEntity = PrivateMessageReactionEntity.fromEventMessage(eventMessage);
+
+    return _db.into(_db.conversationReactionsTable).insert(
+          ConversationReactionsTableData(
+            reactionEventId: reactionEntity.id,
+            createdAt: reactionEntity.createdAt,
+            content: reactionEntity.data.content,
+            messageId: reactionEntity.data.eventId,
+          ),
+          mode: InsertMode.insertOrReplace,
+        );
+  }
+
+  Future<void> insertEventMessage(EventMessage eventMessage) async {
+    await _db.into(_db.eventMessagesTable).insert(
+          EventMessageTableData.fromEventMessage(eventMessage),
+          mode: InsertMode.insertOrReplace,
+        );
+
+    if (eventMessage.kind == PrivateDirectMessageEntity.kind) {
+      await _updateConversationMessagesTable(eventMessage);
+    } else if (eventMessage.kind == PrivateMessageReactionEntity.kind) {
+      switch (eventMessage.content) {
+        case 'received':
+          await _updateConversationMessageAsReceived(eventMessage);
+        case 'read':
+          await _updateConversationMessagesAsRead(eventMessage);
+        default:
+          await _insertConversationReactionsTableData(eventMessage);
+      }
+    }
+  }
+
+  Future<void> _updateConversationMessagesTable(
+    EventMessage eventMessage,
+  ) async {
+    if (eventMessage.kind == PrivateDirectMessageEntity.kind) {
```

Comment:
Use `return fast` or `fail fast` to reduce indentation. Like `if (eventMessage.kind != PrivateDirectMessageEntity.kind) return;`

Reply:
Done